test/post_processing/test_roc_curve_based_post_processing.py

```diff
@@ -3,34 +3,154 @@
+                                                 fairness_metric=DemographicParity())
+
+    with pytest.raises(ValueError, match=PREDICT_BEFORE_FIT_ERROR_MESSAGE):
+        _ = getattr(adjusted_model, predict_method_name)(X, A)
+
+
+def test_both_model_and_estimator_error():
+    with pytest.raises(ValueError, match=EITHER_MODEL_OR_ESTIMATOR_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_unaware_model=ExampleModel(),
+                                        fairness_unaware_estimator=ExampleEstimator(),
+                                        fairness_metric=DemographicParity())
+
+
+def test_no_model_or_estimator_error():
+    with pytest.raises(ValueError, match=MODEL_OR_ESTIMATOR_REQUIRED_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_metric=DemographicParity())
+
+
+def test_metric_not_supported():
+    with pytest.raises(ValueError, match=NOT_SUPPORTED_FAIRNESS_METRIC_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_unaware_model=ExampleModel(),
+                                        fairness_metric=ExampleMetric())
+
+
+def test_not_a_fairness_metric():
+    with pytest.raises(TypeError, match=FAIRNESS_METRIC_EXPECTED_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_unaware_model=ExampleModel(),
+                                        fairness_metric=ExampleNotMetric())
+
+
+@pytest.mark.parametrize("not_estimator", [ExampleNotEstimator1(), ExampleNotEstimator2()])
+def test_not_estimator(not_estimator):
+    with pytest.raises(ValueError, match=MISSING_FIT_PREDICT_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_unaware_estimator=not_estimator,
+                                        fairness_metric=ExampleMetric())
+
+
+def test_not_model():
+    with pytest.raises(ValueError, match=MISSING_PREDICT_ERROR_MESSAGE):
+        _ = ROCCurveBasedPostProcessing(fairness_unaware_model=ExampleNotModel(),
+                                        fairness_metric=ExampleMetric())
+
+
+@pytest.mark.parametrize("X_transform,Y_transform,A_transform",
+                         [(np.array, lambda Y: Y, lambda A: A),
+                          (lambda X: X, np.array, lambda A: A),
+                          (lambda X: X, lambda Y: Y, np.array),
+                          (np.array, np.array, lambda A: A),
+                          (lambda X: X, np.array, np.array),
+                          (np.array, lambda Y: Y, np.array),
+                          (pd.DataFrame, lambda Y: Y, lambda A: A),
+                          (lambda X: X, pd.DataFrame, lambda A: A),
+                          (lambda X: X, lambda Y: Y, pd.DataFrame),
+                          (pd.DataFrame, pd.DataFrame, lambda A: A),
+                          (lambda X: X, pd.DataFrame, pd.DataFrame),
+                          (pd.DataFrame, lambda Y: Y, pd.DataFrame),
+                          (pd.DataFrame, pd.DataFrame, pd.DataFrame)])
```

Comment:
all data frames is no good?

Reply:
Yup, doesn't work right now. I want to first settle on what we want to support before allowing things that don't currently work (and make them work, of course)

Reply:
since Richard's comment points out the same I'll resolve this one.